libyara/include/yara/scanner.h

```diff
@@ -92,7 +92,7 @@ YR_API int yr_scanner_scan_mem_blocks(
     YR_MEMORY_BLOCK_ITERATOR* iterator);
 
 
-YR_API int yr_scanner_scan_mem(
+YR_API int (*yr_scanner_scan_mem)(
```

Comment:
I wouldn't do this, it changes the type of `yr_scanner_scan_mem` which is an exported function part of the API. I think you don't need to completely replace `yr_scanner_scan_mem` in tests, you can use `yr_scanner_scan_mem_blocks` providing your own iterator with custom `first` and `next` functions.

Reply:
Okay. I'll find another way to override the production code biz logic with test code biz logic at test run time. And hopefully the new method will make the clang error on Mac OS go away too :-)

Reply:
@plusvic Okay, so I left yr_scanner_scan_mem() intact so as not to change the API, and implemented the indirect call one level up. Finally I figured out how to appease clang running on Mac OS, and now the build and tests pass :-)